psi4/driver/schema_wrapper.py

```diff
@@ -197,6 +209,142 @@ def _convert_variables(data, context=None, json=False):
     return ret
 
 
+def _convert_basis(basis):
+    """Converts a Psi4 basis object to a QCElemental basis.
+    """
+    centers = []
+    symbols = []
+
+    # Loop over centers
+    for c in range(basis.molecule().natom()):
+        center_shells = []
+        symbols.append(basis.molecule().symbol(c).title())
+
+        # Loop over shells *on* a center
+        for s in range(basis.nshell_on_center(c)):
+            shell = basis.shell(basis.shell_on_center(c, s))
+            if shell.is_pure():
+                htype = "spherical"
+            else:
+                htype = "cartesian"
+
+            # Build the shell
+            coefs = [[shell.coef(x) for x in range(shell.nprimitive)]]
+            exps = [shell.exp(x) for x in range(shell.nprimitive)]
+            qshell = qcel.models.basis.ElectronShell(angular_momentum=[shell.am],
+                                                     harmonic_type=htype,
+                                                     exponents=exps,
+                                                     coefficients=coefs)
```

Comment:
Just a comment here - the coefs you're grabbing are those that are normalized according to the CCA spec, not those found in the basis set spec.  Many (most?) other codes expect a different normalization convention, which is why we have the erd_coefficients around also.